cmake/modules/boards.cmake

```diff
@@ -92,11 +92,28 @@ if(DEFINED ZEPHYR_BOARD_ALIASES)
     set(BOARD_IDENTIFIER ${BOARD_ALIAS_IDENTIFIER}${BOARD_IDENTIFIER})
   endif()
 endif()
+
 include(${ZEPHYR_BASE}/boards/deprecated.cmake)
 if(${BOARD}_DEPRECATED)
   set(BOARD_DEPRECATED ${BOARD} CACHE STRING "Deprecated board name, provided by user")
-  set(BOARD ${${BOARD}_DEPRECATED})
+  parse_board_components(${BOARD}_DEPRECATED BOARD BOARD_DEPRECATED_REVISION BOARD_DEPRECATED_IDENTIFIER)
   message(WARNING "Deprecated BOARD=${BOARD_DEPRECATED} name specified, board automatically changed to: ${BOARD}.")
+  if(DEFINED BOARD_DEPRECATED_REVISION)
+    if(DEFINED BOARD_REVISION)
+      message(FATAL_ERROR
+        "Board revision '${BOARD_REVISION}' conflicts with the new board's revision: "
+        "${BOARD}@${BOARD_DEPRECATED_REVISION}, corresponding to '${BOARD_DEPRECATED}'."
```

Comment:
how exactly / when is this expected to work ?
Can you provide a use-case.

I tried testing with a board using revisions, such as nrf9160dk, and tried to say that the old dk should map to:
```
set(nrf9160dk_nrf9160_DEPRECATED nrf9160dk@0.7.0/nrf9160)
```
I would now expect that if I build for `-DBOARD=nrf9160dk_nrf9160`, then it will be replaced with: `nrf9160dk@0.7.0/nrf9160`, and that part works. But I would also assume that I could build for: `-DBOARD=nrf9160dk_nrf9160@0.7.0`, but that fail with:
```
CMake Error at /projects/github/ncs/zephyr/cmake/modules/boards.cmake:103 (message):
  Board revision '0.7.0' conflicts with the new board's revision:
  nrf9160dk@0.7.0, corresponding to 'nrf9160dk_nrf9160'.
```

What are you trying to prevent with the error ?

Reply:
I disallowed `nrf9160dk_nrf9160@0.7.0` in this case, assuming that if the old, standalone board is now a revision of another board, then there isn't necessarily a clean, transitive mapping of the old board's revision to the new board.

You'd have to deprecate old revisions individually, much like you'd want to deprecate variants.
Sticking to established form:
```
set(nrf9160dk_nrf9160@0.7.0_DEPRECATED nrf9160dk@0.7.0/nrf9160)
```
On the other hand, with a plain board name deprecation:
```
set(nrf9160dk_nrf9160_DEPRECATED nrf9160dk/nrf9160)
```
`nrf9160dk_nrf9160@0.7.0` is allowed, which means the revisions carry over, because they did in HWMv1.

Reply:
> I disallowed nrf9160dk_nrf9160@0.7.0 in this case, assuming that if the old, standalone board is now a revision of another board, then there isn't necessarily a clean, transitive mapping of the old board's revision to the new board.

Agreed, there isn't a clean mapping.
So if this is the intention, then I believe we should change the error message, cause it's not clear what a conflict means, especially if the revision in both cases are the same.

> Sticking to established form (just an example; doesn't work):

Exactly, this doesn't. hence I was concerned if it was actually something related to that you were trying to solve.
So far porting haven't indicated a need for mapping revisions, so let's take such request separately if need arises.

> nrf9160dk_nrf9160@0.7.0 is allowed, which means the revisions carry over, because they did in HWMv1.

Yep, I noticed that 👍